staging/src/k8s.io/apimachinery/pkg/api/validate/schema.go

```diff
@@ -51,3 +58,140 @@ func Required[T comparable](fldPath *field.Path, value T) field.ErrorList {
 	}
 	return nil
 }
+
+// Union verifies that exactly one member of a union is specified.
+//
+// UnionMembership must define all the members of the union.
+//
+// For example:
+//
+//	var abcUnionMembership := schema.NewUnionMembership("a", "b", "b")
+//	func ValidateABC(fldPath, *field.Path, in *ABC) (errs fields.ErrorList) {
+//		errs = append(errs, Union(fldPath, in, abcUnionMembership, in.A, in.B, in.C)...)
```

Comment:
Which field path do we complain about?  All of them?  The parent struct?  The discriminator?

Reply:
I've gone with the parent struct.  The validation of the discriminator is typically handled via enum validation of the discriminator (and is handled separate from the union validation).